src/plots/cartesian/axis_defaults.js

```diff
@@ -142,6 +138,11 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         delete containerOut.zerolinewidth;
     }
 
+    // fill in categories
+    containerOut._initialCategories = axType === 'category' ?
+        orderedCategories(letter, containerOut.categoryorder, containerOut.categoryarray, options.data) :
+        [];
+
     return containerOut;
```

Comment:
@monfera I'm refactoring our top-level supply-defaults functions in #491 , I'll add the info you're requesting to the `Plots.supplyDefaults` [jsDoc](https://github.com/plotly/plotly.js/pull/491/files#diff-ad4f76ccd6044ed16514297078e13b84R447 ) in PR #491 .